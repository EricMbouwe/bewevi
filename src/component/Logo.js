import { Link } from 'react-router-dom';
import logo from '../assets/emptyImage.svg';

const Logo = () => (
  <Link to="/" className="logo main">
    <img className="logo image" src={logo} alt="logo" />
    <div className="logo title">
      vy
    </div>
  </Link>
);

export default Logo;
