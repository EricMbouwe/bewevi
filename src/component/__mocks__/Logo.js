import React from 'react';

const Logo = () => (
  <>
    This is a Logo
  </>
);

export default Logo;
