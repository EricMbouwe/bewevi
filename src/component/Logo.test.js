import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import {
  BrowserRouter, Redirect, Route, Switch,
} from 'react-router-dom';

import Logo from './Logo';

describe('<Logo />', () => {
  it('links to root page', () => {
    render(
      <BrowserRouter>
        <Logo />
        <Redirect to="/second-page" />
        <Switch>
          <Route exact path="/" render={() => <div>Root page</div>} />
          <Route exact path="/second-page" render={() => <div>Second Page</div>} />
        </Switch>
      </BrowserRouter>,
    );
    expect(screen.getByText(/Second Page/i)).toBeInTheDocument();
    expect(screen.getByText(/Ivy/i)).toBeInTheDocument();
    userEvent.click(screen.getByText(/Ivy/i));
    expect(screen.getByText(/Root Page/i)).toBeInTheDocument();
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Logo />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
