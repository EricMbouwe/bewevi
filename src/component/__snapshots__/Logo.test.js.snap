// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> renders correctly 1`] = `
<a
  className="logo main"
  href="/"
  onClick={[Function]}
>
  <img
    alt="logo"
    className="logo image"
    src="emptyImage.svg"
  />
  <div
    className="logo title"
  >
    Ivy
  </div>
</a>
`;
